WowPacketParserModule.V9_0_1_36216/Parsers/QuestHandler.cs

```diff
@@ -566,5 +566,119 @@ public static void HandleQuestGiverCloseQuest(Packet packet)
         {
             packet.ReadInt32<QuestId>("QuestID");
         }
+
+        [Parser(Opcode.SMSG_DISPLAY_PLAYER_CHOICE)]
+        public static void HandleDisplayPlayerChoice(Packet packet)
+        {
+            var choiceId = packet.ReadInt32("ChoiceID");
+            var responseCount = packet.ReadUInt32();
+            packet.ReadPackedGuid128("SenderGUID");
+            var uiTextureKitId = packet.ReadInt32("UiTextureKitID");
+            var soundKitId = packet.ReadUInt32("SoundKitID");
+            packet.ResetBitReader();
+            var questionLength = packet.ReadBits(8);
+            packet.ReadBit("CloseChoiceFrame");
+            var hideWarboardHeader = packet.ReadBit("HideWarboardHeader");
+            var keepOpenAfterChoice = packet.ReadBit("KeepOpenAfterChoice");
+
+            for (var i = 0u; i < responseCount; ++i)
+                ReadPlayerChoiceResponse(packet, choiceId, i, "PlayerChoiceResponse", i);
+
+            var question = packet.ReadWoWString("Question", questionLength);
+
+            Storage.PlayerChoices.Add(new PlayerChoiceTemplate
+            {
+                ChoiceId = choiceId,
+                UiTextureKitId = uiTextureKitId,
+                SoundKitId = soundKitId,
+                Question = question,
+                HideWarboardHeader = hideWarboardHeader,
+                KeepOpenAfterChoice = keepOpenAfterChoice
+            }, packet.TimeSpan);
+
+            if (ClientLocale.PacketLocale != LocaleConstant.enUS)
+            {
+                Storage.PlayerChoiceLocales.Add(new PlayerChoiceLocaleTemplate
+                {
+                    ChoiceId = choiceId,
+                    Locale = ClientLocale.PacketLocaleString,
+                    Question = question
+                }, packet.TimeSpan);
+            }
+        }
+
+        public static void ReadPlayerChoiceResponse(Packet packet, int choiceId, uint index, params object[] indexes)
+        {
+            var responseId = packet.ReadInt32("ResponseID", indexes);
+            var responseIdentifier = packet.ReadInt16("ResponseIdentifier", indexes);
```

Comment:
having ResponseID twice sounds kinda wrong, what is this field about?

cannot check stuff myself rn, will do at a later point

Reply:
No idea, but it has been updated by Shauren on TrinityCore :

https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Server/Packets/QuestPackets.h#L688

I can also confirm this by my own reverse on IDA & by packet parsing : https://i.imgur.com/oTZYJ1Q.png

Reply:
didnt know we already got that in core, we may just leave it like that then